source/WeightedList.h

```diff
@@ -138,6 +139,17 @@ const Type &WeightedList<Type>::Get() const
 
 
 
+template <class Type>
+const Type &WeightedList<Type>::Get(const int index) const
+{
+	if(static_cast<size_t>(index) > choices.size())
```

Comment:
...and in the other place, you're casting size_t to int. I'd suggest using size_t everywhere.